frontend/src/app/register/register.component.ts

```diff
@@ -54,6 +55,12 @@ export class RegisterComponent implements OnInit {
 
   submit() {
     localStorage.removeItem('token_url');
+    const returnUrl = this.route.snapshot.queryParams['returnUrl'];
```

Comment:
could all be one line that sets return_url, no need to delete here, it will always have a value.